### PR TITLE
docs: Add Federated Principal information to iam_policy_document

### PR DESCRIPTION
**Pull request by andyjp94**
This data source already supports Federated access. It just isn't
documented. This change adds that documentation as well as providing
an example of it being used.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
Documented iam_policy_document support for Principals of type "Federated"
```

### CODE REVIEW

File: website/docs/d/iam_policy_document.html.markdown [hunk 1]
```diff
@@ -184,6 +186,10 @@ data "aws_iam_policy_document" "event_stream_bucket_role_assume_role_policy" {
       type        = "AWS"
       identifiers = ["${var.trusted_role_arn}"]
     }
+    principals {
```
Comment (anGie44) [inline website/docs/d/iam_policy_document.html.markdown:189]: Nit: @andyjp94 do you mind adding a newline before this new `principals` block to keep the spacing between blocks consistent?

Comment (andyjp94) [reply website/docs/d/iam_policy_document.html.markdown:189]: sure I can do that

Comment (andyjp94) [reply website/docs/d/iam_policy_document.html.markdown:190]: I've made the changes now, let me know if that is what you expected

Comment (anGie44) [reply website/docs/d/iam_policy_document.html.markdown:190]: looks great! thank you @andyjp94

**Pull request merged**